jupyterhub/spawner.py

```diff
@@ -1254,6 +1270,36 @@ def get_env(self):
 
         return {"keyfile": key, "certfile": cert, "cafile": ca}
 
+    # intended to be run as intermediate process that
+    # calls self.authenticator.pre_exec_fn/post_exec_fn()
+    # after before and after starting the child process
+    # needed by PAMAuthenticator to handle PAM_sessions
+    def start_monitor_process(self, q, cmd, popen_kwargs):
+        if self.authenticator:
+                self.authenticator.pre_exec_fn(self.user, self)
```

Comment:
Does this pre_exec_fn need to run as the user (i.e. after setuid)?

Reply:
Nope. PAM session modules may need elevated privileges to run such mounting drives.